pkg/auth/auth.go

```diff
@@ -2,66 +2,83 @@ package auth
 
 import (
 	"fmt"
+	"time"
 
+	"github.com/Azure/aad-pod-identity/pkg/metrics"
 	"github.com/Azure/aad-pod-identity/version"
 	adal "github.com/Azure/go-autorest/autorest/adal"
 )
 
 const (
-	activeDirectoryEndpoint = "https://login.microsoftonline.com/"
+	activeDirectoryEndpoint                         = "https://login.microsoftonline.com/"
+	adalTokenFromMSIOperationName                   = "adal_token_msi"
+	adalTokenFromMSIWithUserAssignedIDOperationName = "adal_token_msi_userassignedid"
+	adalTokenOperationName                          = "adal_token"
 )
 
+var reporter *metrics.Reporter
+
 // GetServicePrincipalTokenFromMSI return the token for the assigned user
 func GetServicePrincipalTokenFromMSI(resource string) (*adal.Token, error) {
+	begin := time.Now()
 	// Get the MSI endpoint accoriding with the OS (Linux/Windows)
 	msiEndpoint, err := adal.GetMSIVMEndpoint()
 	if err != nil {
+		recordError(adalTokenFromMSIOperationName)
 		return nil, fmt.Errorf("Failed to get the MSI endpoint. Error: %v", err)
 	}
 	// Set up the configuration of the service principal
 	spt, err := adal.NewServicePrincipalTokenFromMSI(msiEndpoint, resource)
 	if err != nil {
+		recordError(adalTokenFromMSIOperationName)
 		return nil, fmt.Errorf("Failed to acquire a token for MSI. Error: %v", err)
 	}
 	// Effectively acquire the token
 	err = spt.Refresh()
 	if err != nil {
+		recordError(adalTokenFromMSIOperationName)
 		return nil, err
 	}
 	token := spt.Token()
-
+	recordDuration(adalTokenOperationName, time.Since(begin))
```

Comment:
should this be `adalTokenFromMSIOperationName`?

Reply:
done.